### ISSUE TITLE
Do you intend on releasing this to Maven Central or JCenter?

### ISSUE THREAD

**Issue by snowe2010** [Carleslc/kotlin-extensions#3]
Like the title asks, do you ever intend on releasing this to Maven Central or JCenter/Bintray?

**Comment by Carleslc** [comment]
Could be interesting, but not very soon. Do you need it? If so, why?

**Comment by snowe2010** [comment]
In general companies will only use a product if it's present in a main repo. My company only uses Maven Central for example. 

It would also be good to split the project up into modules so you could only require what you need instead of polluting your entire top level with extension functions. 

If it's ok with you, I will probably fork this repo, expand it to be modular, and release it to the main repos. I'll probably add a few extension functions of my own as well. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Carleslc** [comment]
Oh, I see. Splitting this project to be modular is a good idea, too. I would be glad if future releases on this repo could be published to Maven Central, so I don't need JitPack repo anymore. Your fork with a PR after that will be enough? I've never published any project to central, if you need my intervention, do you have any good reference about how to do it?

**Comment by snowe2010** [comment]
It is actually quite complex to publish to Maven Central. I read a lot of tutorials before finally getting it to work. The simplest way is to use JCenter to sync, but even then you still have to set up a sonatype account and then ask them if you can publish your repo. If I fork your repo I'll need to change the package name to be my own website, since maven central requires you to own the website (`me.carleslc.kotlin.extensions` will change to something like `com.tylerthrailkill.kotlin.extensions.<module-name>`). I won't be able to publish the project under your name, you'll have to do it. 

The first time I posted to Maven Central it took about a month of work, but I also didn't know what I was doing.

**Comment by Carleslc** [comment]
Ok, package renaming is what I was guessing would be required in the fork. I'll try to publish this repo to Maven Central, but probably it won't be ready until the end of this month or so. Otherwise, if you need that sooner then you can do that fork and all changes you need and publish with your own package name as you said.

**Comment by snowe2010** [comment]
Ok. thank you for the quick response! Good luck! Maven Central is a beast 😅

**Comment by snowe2010** [comment]
Ok. thank you for the quick response! Good luck! Maven Central is a beast 😅

**Comment by Carleslc** [comment]
I've seen your fork, great job with the modules. Since now you're publishing this on your own I think you no longer need that I publish this repo to Maven Central. Currently I'm a bit busy, so I won't publish it soon as I said previously. If anyone really needs this repo released on central repositories please answer this issue or check @snowe2010 fork.

**Comment by snowe2010** [comment]
Yes! Thank you. I haven't gotten to releasing it yet because I previously used TravisCI and am trying to switch off of them to azure, and just haven't had time to get that going. It should be released before this weekend though.

**Comment by bvanseg** [comment]
@snowe2010 I created a repository this fall called KotlinCommons: https://github.com/bvanseg/KotlinCommons, just realized there was another project similar to it (this one).

It's already released on maven central. You're more than welcome to pitch PRs to mine, providing you follow the documentation rules and unit testing rules.

@Carleslc additionally, seeing as we are doing the same thing (kotlin utilities), I would be excited to work on something together with you. If you're interested you can contact my discord listed under my profile name and we can work something out.

**Comment by snowe2010** [comment]
@bvanseg I have had a rough year and I don't remember how much I got done with releasing the code to central. I think I did get it to jcenter though.

**Comment by bvanseg** [comment]
@snowe2010 I see. Sorry to hear you had a rough year. Hopefully it's better now.

My goal is to make a common modular Kotlin utility library that anyone can use. Would you like to collab with me towards that goal?

**Comment by snowe2010** [comment]
@bvanseg for sure!
